### DISCUSSION TITLE
Fetching single item from cache when cache is filled with array

### DISCUSSION THREAD

**Discussion by RubenVanEldik** [Akryum/vue-apollo#1075] [Q&A]
Sorry to create an issue in the main repository, but I couldn't find anything online and didn't receive an answer on the nuxt-apollo discord server.

Is it possible to fetch a singular item from the Apollo cache after I've fetched multiple items on another page?

I have already fetched `stories` on another page, but now I want to fetch a singular item `story(id: $id)` from the cache. So this is a `fetchPolicy: 'cache-only'` query. I have tried a few things, but it looks like Apollo does not know that `stories` is the plural of `story` (which makes sense). Is there anything I can add to let Apollo know that if I ask for a story it can also return the stories item with the same ID?

Would this for example work with something like the IntrospectionFragmentMatcher?

This is quite essential to a project I am working on. If anybody could point me in the right direction I would be eternally grateful!

**Comment by codejie** [comment]
I got the same issue.

While query an array result, I just got the array fill with the first record data, but it is normal when use igraphql.
How can I fetch the RIGHT data?

//////////////////////////////
with tracing and debug, I got the answer that must be with '{fetchPolicy: 'no-cache'}' in Query.
